rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/ExportToCatalogConfirmWindow.java

```diff
@@ -0,0 +1,243 @@
+                                                                                exportNodeSourceToCatalogForm.submit();
+                                                                            } catch (ImportException e) {
+                                                                                String msg = JSONUtils.getJsonErrorMessage(e);
+                                                                                SC.warn("Failed to export node source to catalog:<br>" +
+                                                                                        msg);
+                                                                            } finally {
+                                                                                hideAndDestroy(window);
+                                                                            }
+                                                                        }
+
+                                                                        public void onFailure(Throwable caught) {
+                                                                            String msg = JSONUtils.getJsonErrorMessage(caught);
+                                                                            SC.warn("Failed to fetch configuration of node source " +
+                                                                                    ExportToCatalogConfirmWindow.this.nodeSourceName +
+                                                                                    ":<br>" + msg);
+                                                                            hideAndDestroy(window);
+                                                                        }
+                                                                    });
+
+        hideAndDestroy(this);
+    }
+
+    private RequestCallback fillBucketListWithRequestCallback(ListBox bucketList, Label label) {
+        return new RequestCallback() {
+            @Override
+            public void onResponseReceived(Request request, Response response) {
+                JSONArray bucketsArray = JSONParser.parseStrict(response.getText()).isArray();
+                for (int i = 0; i < bucketsArray.size(); i++) {
+                    JSONObject bucketObject = bucketsArray.get(i).isObject();
+                    bucketList.addItem(bucketObject.get("name").isString().stringValue());
+                }
+                bucketList.setEnabled(true);
+            }
+
+            @Override
+            public void onError(Request request, Throwable t) {
+                String errorMessage = "List buckets from catalog failed";
+                GWT.log(errorMessage, t);
+                label.setContents(errorMessage);
+            }
+        };
+    }
+
+    private void hideAndDestroy(Window window) {
+        window.hide();
+        window.destroy();
+    }
+
+    private class ExportToCatalogHiddenPanel extends VerticalPanel {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 7 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)